### PR TITLE
GenQ PIPELINE: Prereq General Fixes

### PR DESCRIPTION
**Pull request by yfguo**
## Pull Request Description

These are general fixes that GenQ pipeline code needs. Splitting them for reviee and merge. The commits are as follow.

1. FBOX code is removed as it is not currently built and tested because POSIX use iQueue as default. There is no current way to use both FBOX and iQueue.

2-3. There fixes the bugs introduces by previous GenQ PR.

4. Fix one function name prefix. Fix callback registration in am_init---it requires func pointer (which is the function name), but we are passing pointer of a func pointer.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commits are self-contained and do not do two things at once
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] You or your company has a signed contributor's agreement on file with Argonne
* [ ] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

File: src/mpid/ch4/netmod/ofi/ofi_am_impl.c [hunk 1]
```diff
@@ -17,7 +17,7 @@ void MPIDI_OFI_deferred_am_isend_dequeue(MPIDI_OFI_deferred_am_isend_req_t * dre
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_DEFERRED_AM_ISEND_DEQUEUE);
 
     DL_DELETE(MPIDI_OFI_global.deferred_am_isend_q, curr_req);
-    MPL_free(dreq);
+    MPL_free(curr_req);
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_am_impl.c:20]: is this necessary?

File: src/mpid/ch4/netmod/ofi/ofi_am_impl.h [hunk 2]
```diff
@@ -195,11 +197,7 @@ static inline int MPIDI_OFI_progress_do_queue(int vni_idx)
     goto fn_exit;
 }
 
-static inline int MPIDI_OFI_am_isend_long(int rank,
-                                          MPIR_Comm * comm,
-                                          int handler_id,
-                                          const void *am_hdr,
-                                          size_t am_hdr_sz,
+static inline int MPIDI_OFI_am_isend_long(int rank, MPIR_Comm * comm, int handler_id,
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_am_impl.h:200]: I think this API change should be split out to its own commit.

File: src/mpid/ch4/netmod/ofi/ofi_am_impl.h [hunk 3]
```diff
@@ -328,9 +320,9 @@ static inline int MPIDI_OFI_am_isend_short(int rank,
 }
 
 static inline int MPIDI_OFI_deferred_am_isend_enqueue(int rank, MPIR_Comm * comm, int handler_id,
-                                                      size_t am_hdr_sz, const void *buf,
-                                                      size_t count, MPI_Datatype datatype,
-                                                      MPIR_Request * sreq, MPI_Aint data_sz)
+                                                      const void *buf, size_t count,
+                                                      MPI_Datatype datatype, MPIR_Request * sreq,
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_am_impl.h:324]: Same with this API change.

File: src/mpid/ch4/netmod/ofi/ofi_events.c [hunk 4]
```diff
@@ -520,12 +520,13 @@ static int am_isend_event(struct fi_cq_tagged_entry *wc, MPIR_Request * sreq)
...
     int mpi_errno = MPI_SUCCESS;
     MPIDI_OFI_am_header_t *msg_hdr;
+    int incomplete = 0;
 
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_AM_ISEND_EVENT);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_AM_ISEND_EVENT);
 
     msg_hdr = &MPIDI_OFI_AMREQUEST_HDR(sreq, msg_hdr);
-    MPID_Request_complete(sreq);        /* FIXME: Should not call MPIDI in NM ? */
+    MPIR_cc_decr(sreq->cc_ptr, &incomplete);
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_events.c:529]: This goes against my understanding of how to properly complete requests. It's true that we manipulate the cc directly when incrementing, but it should be OK to always call a completion function when decrementing it. Is the issue that the completion handling function will be called prematurely?

File: src/mpid/ch4/netmod/ofi/ofi_am_impl.c [hunk 5]
```diff
@@ -39,6 +40,10 @@ int MPIDI_OFI_deferred_am_isend_issue(MPIDI_OFI_deferred_am_isend_req_t * dreq)
     need_packing = dt_contig ? false : true;
     send_buf = (char *) dreq->buf + dt_true_lb;
 
+    data_lt_eager_threshold =
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_am_impl.c:43]: what is "lt"?

Comment (raffenet) [reply src/mpid/ch4/netmod/ofi/ofi_am_impl.c:43]: Oh, "less than". Can we change the name of this to `bool do_eager`?

Comment (raffenet) [reply src/mpid/ch4/netmod/ofi/ofi_am_impl.c:43]: Another nit, this is not a really a fix and could be split out.

Comment (yfguo) [reply src/mpid/ch4/netmod/ofi/ofi_am_impl.c]: True, this is not a real fix. It is more of avoiding long calculation lines.

Comment (yfguo) [reply src/mpid/ch4/netmod/ofi/ofi_am_impl.c]: I have renamed it.
